platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrClientFactory.java

```diff
@@ -13,21 +13,22 @@
  */
 package org.codice.solr.factory;
 
-import java.util.concurrent.Future;
-import org.apache.solr.client.solrj.SolrClient;
+import org.codice.solr.client.solrj.SolrClient;
 
 /** Interface implemented by factory classes used to create new {@link SolrClient} instances. */
 public interface SolrClientFactory {
-
   /**
-   * Requests the creation of a new {@code SolrClient} for a specific Solr core name. <br>
-   * Note that {@link Future#get()} will return {@code null} if a {@link SolrClient} could not be
-   * created either immediately or after a retry period determined by the implementing class.
-   * Clients of this class should consider implementing retry logic if needed based on that return
-   * value.
+   * Requests the creation of a new {@code SolrClient} for a specific Solr core name.
+   *
+   * <p><i>Note:</i> The client returned might not yet be available (see {@link
```

Comment:
Since this is the place where you get you Solr client, I think it is sound to warn you about that.
As for the SolrClient, I think there are already warnings of the sort but I would be ok to copy it there at the top as well.

Reply:
👍